From ac75d372d838c7554c8c3545700b09567e3f6d4e Mon Sep 17 00:00:00 2001
From: sai surya teja <sssteja@habana.ai>
Date: Wed, 11 Jan 2023 05:40:13 +0200
Subject: [PATCH] [SW-114788] gaudi2: security: reduce hbw rr address range for SRAM

With the recent decision to block reserved regions in HBW to avoid
PSOC_RAZW interrupts when a user tries to access unreserved region, full
address HBW RRs are used for protecting reserved regions above PCIE_64PB
(i.e 0x0100000000000000-0x1000007FF4000000) and above HBM (i.e
0x1002000000000000 - 0xFF00000000000000)

Due to this, we need to use short RR for SRAM and since short RR has
an address format of 0x1000_<36bits>000, we cannot protect the last 4K
sector of SRAM. Hence adjusting the max address of HBW RR configuration
used for SRAM.

Change-Id: I39dbf445f6754d82d7c5a24d57fecc5606b333fc
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 1470a44..de2383d 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -153,7 +153,8 @@
 	.rr_en = mmDCORE0_HMMU0_MMU_DDR_RANGE_REG_ENABLE,
 };
 
-#define SRAM_END	(SRAM_BASE_ADDR + SRAM_SIZE)
+#define SRAM_END		(SRAM_BASE_ADDR + SRAM_SIZE)
+#define SRAM_END_EXC_TOP4K	(SRAM_END - 4096) /* Short HBW RR can't protect last 4K of SRAM */
 
 static struct hl_sec_dyn_pcie_rr_cfg preboot_sec_dyn_cfg[] = {
 	[SRAM_BGN_TO_IMG] = {
@@ -992,10 +993,10 @@
 
 	/*
 	 * Block Reserved space to HBM
-	 * 0x1000FFFFFD000000 - 0x1001000000000000
+	 * 0x1000FFFFFD000000 - 0x1000FFFFFFFFF000
 	 */
 	min = SRAM_RSVD_BASE;
-	max = SRAM_BASE_ADDR + SRAM_SIZE;
+	max = SRAM_END_EXC_TOP4K;
 	hl_secure_hbw_rr(HL_SEC_HBW_SRAM, min, max, rr_sel);
 
 	/*
